internal/proxy/task_search.go

```diff
@@ -302,7 +302,7 @@ func (t *searchTask) Execute(ctx context.Context) error {
 	}
 
 	err := executeSearch(WithCache)
-	if err == errInvalidShardLeaders || funcutil.IsGrpcErr(err) {
+	if err == errInvalidShardLeaders {
```

Comment:
It's better to check this with errors.Is(err, errInvalidShardLeaders) which also works if err is wrapped